mavedb_mapping/blat_alignment.py

```diff
@@ -0,0 +1,369 @@
+
+def get_gene_data(gsymb: str)->tuple:
+    """
+    Searches through QueryHandler using obtained gene symbol.
+
+    Parameters
+    ----------
+        gsymb: str
+            Gene symbol
+
+    Returns:
+    --------
+        tuple:
+            source_dict: dictionary of sources with their index in temp
+            temp: QueryHandler search output
+    
+    """
+    if not gsymb:
+        return (None, None)
+    temp = qh.search(gsymb).source_matches
+    source_dict = {}
+    for i in range(len(temp)):
+        source_dict[temp[i].source] = i
+    return temp, source_dict
+
+
+def get_hgnc_accession(temp, source_dict:dict):
+    """Function that obtains HGNC accession id (like hgnc:12345)"""
+    if temp == "NA":
+        return "NA"
+    accession = temp[source_dict["HGNC"]].records[0].concept_id
+    return accession
+
+
+def get_sequence_interval(records):
+    """
+    Helper function to obtain gene data
+
+    """
+    for record in records:
+        for location in record.locations:
+            if location.interval.type == "SequenceInterval":
+                start = location.interval.start.value
+                end = location.interval.end.value
+                loc_list = {"start": start, "end": end}
+                return loc_list
+    return None
+
+
+def return_gene_data(return_chr: bool, temp, source_dict:dict):
+    """
+    Parameters
+    ----------
+        return_chr :bool
+           If True, returns chromosome information.
+
+        source_dict: dict
+            dictionary of sources with their index in temp
+
+        temp: QueryHandler search output
+
+
+    Returns:
+    --------
+        str:
+            If return_chr is True, returns the chromosome value as a string.
+            If gene symbol cannot be extracted, returns 'NA'.
+        OR
+        dict:
+            If gene symbol can be extracted and return_chr is False
+
+    """
+    if not temp:
+        return "NA"
+
+    if "HGNC" in source_dict and return_chr == True:
+        chrom = temp[source_dict["HGNC"]].records[0].locations[0].chr
+        return chrom
+    if (
+        source_dict.get("Ensembl") is not None
+        and return_chr == False
+        and len(temp[source_dict["Ensembl"]].records) != 0
+    ):
+        loc_list = get_sequence_interval(temp[source_dict["Ensembl"]].records)
+        if loc_list:
+            return loc_list
+
+    if (
+        source_dict.get("NCBI") is not None
+        and return_chr == False
+        and len(temp[source_dict["NCBI"]].records) != 0
+    ):
+        loc_list = get_sequence_interval(temp[source_dict["NCBI"]].records)
+        if loc_list:
+            return loc_list
+
+    return "NA"
+
+
+def extract_blat_output(dat: dict):
+    """
+    Runs a BLAT Query and returns the output
+
+    Parameters
+    ----------
+        dat: dict
+            Dictionary containing data required for mapping.
+
+    Returns:
+    --------
+        BLAT Output
+    """
+    blat_file = open("blat_query.fa", "w")
```

Comment:
It could be helpful to make the query filename an argument that could be optionally given to this function:

```suggestion
def extract_blat_output(dat: dict, query_filename: str = "blat_query.fa"):
    """
    Runs a BLAT Query and returns the output

    Parameters
    ----------
        dat: dict
            Dictionary containing data required for mapping.
        blat_query: str
            Filename for BLAT query file to save

    Returns:
    --------
        BLAT Output
    """
    blat_file = open(query_filename, "w")
```